railgun/pkg/config/config.go

```diff
@@ -197,3 +207,42 @@ func (c *Config) GetKubeClient() (client.Client, error) {
 	}
 	return client.New(conf, client.Options{})
 }
+
+func (c *Config) ConfigKongService(ctx context.Context) error {
+	// deployment configuration
+	adminApiService := os.Getenv("CONTROLLER_KONG_ADMIN_PUBLISH_SERVICE")
+	if adminApiService == "" {
+		// parameter configuration
+		if len(c.KongAdminURL) > 0 && c.KongAdminURL != "http://localhost:8001" {
+			return nil
+		}
+		return fmt.Errorf(" kong admin api url %s should not be localhost.", c.KongAdminURL)
+	}
+
+	kubeCfg, err := c.GetKubeconfig()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve kubeconfig. err %v", err)
```

Comment:
to be more consistent with the rest of the code, we should do error wrapping: 
```suggestion
		return fmt.Errorf("failed to retrieve kubeconfig: %w", err)
```